### PR TITLE
Refactor timepoint retrieval & conversion to signed entities

### PR DESCRIPTION
**Pull request by Alenar**
## Content

Following changes introduced in #1727 the conversion of time points to signed entities became more cumbersome. Instead of only using a `CardanoNetwork` it also needed now the newly introduced `CardanoTransactionsSigningConfig`.

This PR make this easier by delegating the configuration to a new in memory type: `SignedEntityConfig`, that is created at configuration time and is in the application dependency container.

This type also takes responsibility, previously in the aggregator `Configuration`, to list allowed signed entity types from a time point. It also hold statically the list of default types that should always be allowed.

Also in this PR: Rename `TimePointProvider` to `TickerService` and remove the conflicting service, named the same, from the aggregator.
Those two were doing the same things and comments were in the code to remove the `TimePointProvider` in favor of the `TickerService`, but the former continued to evolved even if the later api & naming were preferred. So this PR fixes this contradiction by "fusing" them together.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

## Issue(s)
Relates to #1697 and #1727

### CODE REVIEW

**Review by sfauvel — APPROVED** [review]
LGTM

File: mithril-aggregator/src/configuration.rs [hunk 1]
```diff
@@ -272,21 +272,25 @@ impl Configuration {
             .map(|limit| if limit > 3 { limit as u64 } else { 3 })
     }
 
-    /// Create the deduplicated list of allowed signed entity types discriminants.
+    /// Deduce the [SignedEntityConfig] from this configuration.
+    pub fn deduce_signed_entity_config(&self) -> StdResult<SignedEntityConfig> {
```
Comment (jpraynaud) [inline mithril-aggregator/src/configuration.rs:276]: Maybe this is more accurate:
```suggestion
    pub fn compute_signed_entity_config(&self) -> StdResult<SignedEntityConfig> {
```

File: mithril-aggregator/src/dependency_injection/containers.rs [hunk 2]
```diff
@@ -45,8 +46,11 @@ pub struct DependencyContainer {
     /// Configuration structure.
     pub config: Configuration,
 
+    /// Signed entity configuration.
+    pub signed_entity_config: SignedEntityConfig,
+
     /// SQLite database connection
-    /// This is not a real service but is is needed to instantiate all store
+    /// This is not a real service but is needed to instantiate all store
     /// services. Shall be private dependency.
```
Comment (jpraynaud) [inline mithril-aggregator/src/dependency_injection/containers.rs:54]:
```suggestion
    /// This is not a real service but it is needed to instantiate all store
    /// services. Should be a private dependency.
```

File: mithril-aggregator/src/configuration.rs [hunk 3]
```diff
@@ -298,32 +302,6 @@ impl Configuration {
 
         Ok(all_discriminants)
     }
-
-    /// Create the deduplicated list of allowed signed entity types.
-    ///
-    /// By default, the list contains the MithrilStakeDistribution and the CardanoImmutableFilesFull.
-    /// The list can be extended with the configuration parameter `signed_entity_types`.
-    /// The signed entity types are defined in the [SignedEntityTypeDiscriminants] enum.
-    /// The signed entity types are discarded if they are not declared in the [SignedEntityType] enum.
```
Comment (jpraynaud) [inline mithril-aggregator/src/configuration.rs:307]: This comment could be added to the `parse_signed_entity_types_discriminants` function IMO

**Review by jpraynaud — APPROVED** [review]
LGTM :+1: 

I left few comments below.

**Review by dlachaume — APPROVED** [review]

File: mithril-aggregator/src/configuration.rs [hunk 4]
```diff
@@ -132,6 +132,9 @@ pub struct Configuration {
 
     /// Signed entity types parameters (discriminants names in an ordered, case-sensitive, comma
     /// separated list).
+    ///
+    /// The values MithrilStakeDistribution and CardanoImmutableFilesFull are appended automatically
+    /// to the list.
```
Comment (jpraynaud) [inline mithril-aggregator/src/configuration.rs:137]:
```suggestion
    ///
    /// The values MithrilStakeDistribution and CardanoImmutableFilesFull are prepended automatically
    /// to the list.
```

**Pull request merged**